### PR TITLE
[NO-TICKET] Clean up lint warnings

### PR DESCRIPTION
**Pull request by dmethvin-gov**
## Summary

Clean up the easy-to-fix lint warnings. They generally fall into a few categories:
* SCSS syntax that stylelint didn't recognize -- Added SCSS directives to the allowed list
* Things `import`ed but never used in the file -- Removed the imports
* Multiple `import`s for the same file -- Combined the imports on one line
* Function arguments never used -- Removed the args
* Functions with empty bodies -- Had them return `null` which is fine
* dangerouslySetInnerHtml -- Disabled the warning if the source string seemed safe
* Other things: Explanation comments in the PR below

   **Before**: 15 stylelint warnings, 113 eslint warnings
   **After**: 5 stylelint warnings, 68 eslint warnings

The remaining warnings fall into a few categories as well. They involve non-cosmetic changes to the code or non-trivial changes to dependencies, so I didn't attempt them. In particular there are a lot of these:

* Missing dependencies in `useEffect` functions - These can cause erratic behavior so it would be good to analyze and fix them. These sort of changes often trade one sort of (perhaps undiscovered) bug for another.
* Things that shouldn't give warnings. For example, eslint is configured to allow unused variables when defined along with `...rest` declarations but it doesn't seem to be working. It might be fixed by an ESLint upgrade, but when I tried doing that I encountered conflicts. So I left all that alone for now.

## How to test

There should be no functional difference. All existing unit and snapshot tests pass but one, and I can't see the difference in it but it might be due to an extra space after removing the unnecessary fragment.

[webkit] stories.test.ts:70:13 › Components/Icons/Available Icons › with medicare theme › matches snapshot

## Checklist

- [ ] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [ ] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [ ] Selected appropriate `Impacts`, multiple can be selected.
- [ ] Selected appropriate release milestone

### CODE REVIEW

File: packages/design-system/src/components/Table/Table.stories.tsx
Comment (dmethvin-gov) [inline packages/design-system/src/components/Table/Table.stories.tsx:102]: Eliminates a "using index as a key" lint warning.

File: packages/design-system/src/components/analytics/getAnalyticsContentFromRefs.ts
Comment (dmethvin-gov) [inline packages/design-system/src/components/analytics/getAnalyticsContentFromRefs.ts:5]: This arg isn't used in the function.

File: packages/design-system/src/components/Pagination/Pagination.tsx
Comment (dmethvin-gov) [inline packages/design-system/src/components/Pagination/Pagination.tsx:198]: Since `map` isn't returning anything, `forEach` is the right choice here.

File: packages/design-system/src/components/Icons/icons.stories.tsx
Comment (pwolfert) [inline packages/design-system/src/components/Icons/icons.stories.tsx:199]: 👍

File: packages/design-system/src/components/Pagination/Pagination.tsx
Comment (pwolfert) [reply packages/design-system/src/components/Pagination/Pagination.tsx:198]: Nice, yep, that makes sense 👍

File: packages/design-system/src/components/Table/Table.tsx
Comment (pwolfert) [inline packages/design-system/src/components/Table/Table.tsx:6]: Nice, no more lodash here 🎉

File: packages/design-system/src/components/analytics/getAnalyticsContentFromRefs.ts
Comment (pwolfert) [reply packages/design-system/src/components/analytics/getAnalyticsContentFromRefs.ts:5]: Looks like it isn't used anymore, but there is a lingering instance of it being called with the second parameter that needs to be updated. Right now it fails to build.

File: packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx
Comment (pwolfert) [inline packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:1]: We held back on removing these React imports because of a problem we found somewhere with backwards compatibility. It doesn't look like there's any problem with removing them from the story files, but there may be an issue with removing it from the ones we export in our package. I need to do a little more digging.

**Review by pwolfert — COMMENTED** [review]
Hey, thanks for cleaning these up! It looks like most of this will work, but there's a build error to fix, and there's some research I need to do on the React import removals.

**Comment by pwolfert** [comment]
not a wolf

File: packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx
Comment (dmethvin-gov) [reply packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:1]: I noticed that as well with React imports in the main code (not Storybook). I think it probably has something to do with the new JSX transform introduced a few years ago. It's possible the current build setup may be using the old way, which required React in scope even if it wasn't being explicitly used. 

https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html

File: packages/design-system/src/components/analytics/getAnalyticsContentFromRefs.ts
Comment (dmethvin-gov) [reply packages/design-system/src/components/analytics/getAnalyticsContentFromRefs.ts:5]: Somehow I missed that! I must have done the build before I saved this file. I really do appreciate that the project has a thorough set of unit and snapshot tests, it makes it much easier for newbs like me to change things and know if we're breaking something.

File: packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx
Comment (dmethvin-gov) [reply packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:1]: I just checked and yes, it's using the old JSX transform. Since the min peerDependency for React is 17.0.0 it should be possible to upgrade without a major version bump. Several of the build dependencies are almost 2 years old so it might be a good idea to freshen them up as well. If you want I can take a look after I look into the react-day-picker+date-utils thing.

Comment (pwolfert) [reply packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:1]: I actually did not do a good job of making sure that peer dependency version stayed where it needed to. We're technically still supporting React 16 projects, mostly on the medicare side. Hopefully this year we'll be able to drop React 16. For that reason, I don't think we're ready for this. Thank you for the offer to look into it. If you really want to, it might be good to figure out everything that we will need to and get to change when we do finally drop React 16, even if we can't merge a PR right now.

Comment (dmethvin-gov) [reply packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:1]: For the ones that I removed, there was no JSX being rendered in the file so React wasn't needed regardless. It looks like the medicare.gov home page is on 17.0.2, according to React devtools? I used to have visibility into the medicare and healthcare repos but they moved them late last year.

**Comment by dmethvin-gov** [comment]
I updated this to remove componentName everywhere. The failure is probably the one previous snapshot failure, but I don't have visibility into the Jenkins jobs to see what failed. If it's something else, let me know.

File: packages/design-system/src/components/HelpDrawer/useHelpDrawerAnalytics.ts
Comment (pwolfert) [inline packages/design-system/src/components/HelpDrawer/useHelpDrawerAnalytics.ts:42]: Your changes here made me realize that in [this change](https://github.com/CMSgov/design-system/pull/1944) not only did I fail to remove the unused `componentName` property, but I also mislabeled this component through copypasta. Since this is in a fork, would you mind [updating this string for me](https://github.com/CMSgov/design-system/blob/main/packages/design-system/src/components/HelpDrawer/useHelpDrawerAnalytics.ts#L27) to reflect that it's for the help drawer?

File: packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx
Comment (pwolfert) [reply packages/ds-healthcare-gov/src/components/ThirdPartyExternalLink/ThirdPartyExternalLink.tsx:1]: Oh, I understand. I didn't realize that there was no JSX in those files. Carry on!

**Review by pwolfert — APPROVED** [review]
I have one request, but LGTM otherwise!

Currently investigating some sort of instability in my storybook-docs tests, which is what is currently failing your CI tests. (You can see it locally if you're curious with `yarn build:storybook && yarn test:browser:storybook-docs`.)

File: packages/ds-medicare-gov/src/components/HelpDrawer/HelpDrawer.stories.tsx
Comment (pwolfert) [inline packages/ds-medicare-gov/src/components/HelpDrawer/HelpDrawer.stories.tsx:62]: I went down a rabbit hole of cleaning up these stories trying to get the arg tables to show up. Right now this medicare-specific story shows either a table with bogus args or no table at all. Both the main `HelpDrawer` and this Medicare-specific one need work, but I'm spending too many cycles on it right now and need to move on. Would you mind reverting this module's changes for now so the CI tests pass, and you can merge?

Comment (pwolfert) [reply packages/ds-medicare-gov/src/components/HelpDrawer/HelpDrawer.stories.tsx:62]: I'll create a follow-up PR at a future date

Comment (pwolfert) [reply packages/ds-medicare-gov/src/components/HelpDrawer/HelpDrawer.stories.tsx:62]: @dmethvin-gov, the current CI failure is due to these modules changes. If you revert this module to what's on `main`, they should pass. I'll clean these up at some point.

Comment (dmethvin-gov) [reply packages/ds-medicare-gov/src/components/HelpDrawer/HelpDrawer.stories.tsx]: That is super strange, since they aren't used. However, it did indeed fix the problem locally?

Comment (pwolfert) [reply packages/ds-medicare-gov/src/components/HelpDrawer/HelpDrawer.stories.tsx]: [Here's the follow-up](https://github.com/CMSgov/design-system/pull/2902)

**Pull request merged**
